### DISCUSSION TITLE
Error: [11:39:55.591] The build failed because the process exited too early. This probably means the system ran out of memory or someone called `kill -9` on the process.
[11:39:55.608] Error: Command "npm run build" exited with 1
[11:39:55.908] BUILD_UTILS_SPAWN_1: Command "npm run build" exited with 1

### DISCUSSION THREAD

**Discussion by dpvasani** [vercel/next.js#51880] [Help]
How To Solve This error?

**Comment by icyJoseph** [comment]
Well, it either timeout out or really did run out of memory. 

Where is this happening? Vercel? or locally? other hosting provider?

How many pages are you building? what kind of data is it handling?

**Comment by dpvasani** [comment]
PS D:\Dp Main Portfolio> npx run build
Watching D:\Dp Main Portfolio and all sub-directories not excluded by your .gitignore. Will not monitor dotfiles.
D:\Dp Main Portfolio\node_modules\run\run.js:127
        if (minimatch(file, pattern)) {
            ^

TypeError: minimatch is not a function
    at D:\Dp Main Portfolio\node_modules\run\run.js:127:13
    at Array.forEach (<anonymous>)
    at parseFolder (D:\Dp Main Portfolio\node_modules\run\run.js:113:9)
    at Object.<anonymous> (D:\Dp Main Portfolio\node_modules\run\run.js:63:12)
    at Module._compile (node:internal/modules/cjs/loader:1218:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Module._load (node:internal/modules/cjs/loader:922:12)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)

Node.js v18.13.0
PS D:\Dp Main Portfolio>

**Reply by icyJoseph** [reply]
Running `npx run build` will install a binary called `run` you don't want that! You need to do `npm run build`, or `npx next build`-

**Reply by dpvasani** [reply]
Still not solved
While npm run build 

Error:[11:39:55.591] The build failed because the process exited too early. This probably means the system ran out of memory or someone called `kill -9` on the process.
[11:39:55.608] Error: Command "npm run build" exited with 1
[11:39:55.908] BUILD_UTILS_SPAWN_1: Command "npm run build" exited with 1

**Reply by icyJoseph** [reply]
Yes, but now we are back on track. 

- I wonder if you have an unhandled promise rejection.
- How large is this project?
- Could it be that the machine doesn't have enough RAM?

**Comment by dpvasani** [comment]
project size after node module is aroung 200 mb
and system rem is 24 gb there is no issues on ram 
same project if i do in my local compute working fine but in vercel that is giving error

**Comment by dpvasani** [comment]
this is project:https://dpvasani-57.netlify.app/
i done build folder on my coputer and than i uploaded it on netlify
if i do from github their is error throwing

**Reply by icyJoseph** [reply]
Do you have the same Node.js version locally and on Vercel?

Is it pulling data from a 3rd party endpoint? Is that endpoint available to the build step on Vercel?

**Comment by bloodykheeng** [comment]
try to increase your swap memory
for me my issue was that i couldnt npm run build on linux server
out of the blue the terminal was giving me message killed

after some research i found out that my swap memory was 500mb and i extended it to 1gb so the build ran well now
my server was 4gb

to check for swap
cd into root
then type

```
free
```
that will show u the available swap memory

the commands below will increase it to 1gb

```
  sudo /bin/dd if=/dev/zero of=/var/swap.1 bs=1M count=1024
  sudo /sbin/mkswap /var/swap.1
  sudo /sbin/swapon /var/swap.1
```

**Comment by aashay4123** [comment]
> **Hey!**  
> 
> I faced a similar issue before — when your build crashes with an error like "process exited too early" or "npm run build exited with 1," it's usually because the server (like Vercel or Netlify) runs out of memory during the heavy build phase.
> 
> If you're building locally and you see it crash too, you can increase Node.js memory limits by running:
> 
> ```
> NODE_OPTIONS="--max_old_space_size=4096" npm run build
> ```
> 
> This allows Node to use up to 4GB of RAM during the build. You can adjust `4096` to a higher number if needed depending on your machine.
> 
> Also, keep in mind that free hosting plans (like on Vercel or Netlify) usually offer very limited RAM (around 1-2GB), so for bigger apps, it might fail on their servers even if it builds fine locally.  
> In that case, optimizing your build (smaller images, less heavy operations) or upgrading the plan helps a lot!
> 
> Hope this helps!
